### DISCUSSION TITLE
Support Vue getters

### DISCUSSION THREAD

**Discussion by henribru** [TanStack/query#5461] [Ideas] [answered]
Vue 3.3 introduced a concept of getters, similar to a `computed` except you don't actually wrap the function in a `computed`, making it simpler and also more performant in simple cases:

https://blog.vuejs.org/posts/vue-3-3#:~:text=%3C/script%3E-,Better%20Getter%20Support%20with%20toRef%20and%20toValue,-toRef%20has%20been

I think all that's really needed to support this in Tanstack Query is to use [toValue](https://vuejs.org/api/reactivity-utilities.html#tovalue) instead of `unref`. The `toValue` implementation is very simple so I figure we can just copy it so we don't have to depend on Vue 3.3. Would such a contribution be welcome? The one downside I see is that if you have a function in your query key then this is actually a breaking change because it would be treated as a getter. Not sure if that's something people do :thinking:

**Comment by DamianOsipiuk** [comment, marked as answer]
This seems to be interesting, but could be tricky to implement, due to the fact that we accept a set of functions in `queryOptions`, like `queryFn`, `select` etc and we do not actually want to evaluate them during unwrapping.
Additionally we need to deeply unwrap the options to account for nested refs.

**Reply by henribru** [reply]
You're right, it's definitely troublesome outside of the query key. I suppose it would be a bit too inconsistent to only apply it there. Oh well, I can live with some extra `computed`s
